server/src/main/java/org/hl7/davinci/endpoint/rems/database/requirement/Requirement.java

```diff
@@ -0,0 +1,83 @@
+package org.hl7.davinci.endpoint.rems.database.requirement;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.hl7.davinci.endpoint.rems.database.drugs.Drug;
+import org.hl7.davinci.endpoint.rems.database.fhir.RemsFhir;
+
+import javax.persistence.*;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+@Entity
+@Table(name = "requirement")
+public class Requirement {
+    @Id
+    @GeneratedValue
+    @Column(name = "id", nullable = false)
+    private Integer id;
+
+    @Column(name = "timestamp", nullable = false)
+    private String timestamp;
+
+    @Column(name = "completed", nullable = false)
+    private boolean completed ;
+
+    @Column(name = "description", nullable = true)
+    private String description;
+
+    // this should be a reference to a FHIR resource
+    @JoinColumn(name = "requirement", nullable = false)
+    @OneToOne
+    private RemsFhir requirement;
```

Comment:
If you mean the comment, I think that's a holdover that I forgot to delete.  RemsFhir is open to changes but it is the FHIR database here.

Reply:
Perhaps I'm missing some context about the plan for this API, but wouldn't we want it to to be a FHIR API serving FHIR resources rather than non-FHIR objects with FHIR resources encapsulated as strings? And rather than a Drug it seems to me the top level object should be a REMS which can list or reference the drug(s) it's for, and the REMS object could be represented with a FHIR profile for the API's sake. I realize that the API design is out of the scope of this ticket, but it would influence the database design. Thoughts? I very well may be misunderstanding something important.

Reply:
a lot of these things aren't super clear right now, the database will evolve as we develop the REMS admin and the workflow post-pharmacy system.   The database as it stands is using FHIR as a backdrop, it can store anything we want.  It's not clear right now whether a FHIR-based approach will be the most effective.